runtime/queries/tcl/highlights.scm

```diff
@@ -0,0 +1,153 @@
+
+(comment) @comment
+
+(command name: (simple_word) @function)
+
+"proc" @keyword.function
+
+(procedure
+  name: (_) @variable
+)
+
+(set (simple_word) @variable)
+
+(argument
+  name: (_) @variable.parameter
+)
+
+((simple_word) @variable.builtin
+               (#any-of? @variable.builtin
+                "argc"
+                "argv"
+                "argv0"
+                "auto_path"
+                "env"
+                "errorCode"
+                "errorInfo"
+                "tcl_interactive"
+                "tcl_library"
+                "tcl_nonwordchars"
+                "tcl_patchLevel"
+                "tcl_pkgPath"
+                "tcl_platform"
+                "tcl_precision"
+                "tcl_rcFileName"
+                "tcl_traceCompile"
+                "tcl_traceExec"
+                "tcl_wordchars"
+                "tcl_version"))
+
+
+"expr" @function.builtin
+
+(command
+  name: (simple_word) @function.builtin
+  (#any-of? @function.builtin
+   "cd"
+   "exec"
+   "exit"
+   "incr"
+   "info"
+   "join"
+   "puts"
+   "regexp"
+   "regsub"
+   "split"
+   "subst"
+   "trace"
+   "source"))
+
+(command name: (simple_word) @keyword
+         (#any-of? @keyword
+          "append"
+          "break"
+          "catch"
+          "continue"
+          "default"
+          "dict"
+          "error"
+          "eval"
+          "global"
+          "lappend"
+          "lassign"
+          "lindex"
+          "linsert"
+          "list"
+          "llength"
+          "lmap"
+          "lrange"
+          "lrepeat"
+          "lreplace"
+          "lreverse"
+          "lsearch"
+          "lset"
+          "lsort"
+          "package"
+          "return"
+          "switch"
+          "throw"
+          "unset"
+          "variable"))
+
+[
+ "error"
+ "namespace"
+ "on"
+ "set"
+ "try"
+ ] @keyword
+
+(unpack) @operator
+
+[
+ "while"
+ "foreach"
+ ; "for"
+ ] @keyword.control.repeat
+
+[
+ "if"
+ "else"
+ "elseif"
+ ] @keyword.control.conditional
+
+[
+ "**"
+ "/" "*" "%" "+" "-"
+ "<<" ">>"
+ ">" "<" ">=" "<="
+ "==" "!="
+ "eq" "ne"
+ "in" "ni"
+ "&"
+ "^"
+ "|"
+ "&&"
+ "||"
+ ] @operator
+
+(variable_substitution) @variable
+(quoted_word) @string
+(escaped_character) @constant.character.escape
+
+[
+ "{" "}"
+ "[" "]"
+ ";"
+ ] @punctuation.delimiter
+
+((simple_word) @constant.numberic
+               (#lua-match? @constant.numberic "^[0-9]+$"))
```

Comment:
```suggestion
               (#match? @constant.numberic "^[0-9]+$"))
```

`#lua-match?` is specific to neovim but I believe this regex is already rust-regex compatible. We can just switch to the `#match?` predicate